llvm/include/llvm/ADT/AddressRanges.h

```diff
@@ -83,17 +85,18 @@ template <typename T> class AddressRangesBase {
   typename Collection::const_iterator begin() const { return Ranges.begin(); }
   typename Collection::const_iterator end() const { return Ranges.end(); }
 
-  const T &operator[](size_t i) const {
-    assert(i < Ranges.size());
-    return Ranges[i];
+  const T &operator[](size_t I) const {
+    assert(I < Ranges.size());
+    return Ranges[I];
   }
 
-  bool operator==(const AddressRangesBase<T> &RHS) const {
+  bool operator==(const AddressRangesBase &RHS) const {
     return Ranges == RHS.Ranges;
   }
```

Comment:
This should probably be a friend or a free functions, but fine to keep as-is

Reply:
I have kept it as-is if you don't mind.